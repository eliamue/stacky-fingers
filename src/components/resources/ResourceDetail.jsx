import React from 'react';
import styles from '../styles/Detail.css';
import { useParams } from 'react-router-dom';
import { useOneResource } from '../../hooks/useOneResource';

const spinner = 'https://i.redd.it/o6m7b0l6h6pz.gif';

// const always = '*Available 24/7';

const ResourceDetail = () => {
  const { id } = useParams();
  const { loading, resource } = useOneResource(id);
  const {
    name,
    category,
    description,
    address,
    city,
    us_state,
    zip,
    phone,
    text,
    website,
    email,
    is247,
  } = resource;

  if (loading) return <img src={spinner} alt="Loading" />;

  return (
    <div className={styles.srcdeets}>
      <h1>{name}</h1>

      <h3>Category: {category}</h3>

      <h4 className={styles.desc}>{description}</h4>

      <section className={styles.location}>
        <h3>Location</h3>
        <p>{address}</p>
        <p>
          {city}, {us_state} {zip}
        </p>
      </section>

      <section className={styles.contact}>
        <h3>Contact</h3>
        <p>Phone number: {phone}</p>
        <p>Text number: {text}</p>
        <p>
          <a href="true" src={website} alt={name} className={styles.website}>
            {website}
          </a>
        </p>
        <p>{email}</p>
        <p>Available 24/7? {is247}</p>
      </section>
    </div>
  );
};

export default ResourceDetail;
