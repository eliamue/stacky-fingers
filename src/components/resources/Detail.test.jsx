import React from 'react';
import { render, screen } from '@testing-library/react';
import ResourceDetail from './ResourceDetail.jsx';
import { MemoryRouter } from 'react-router-dom';

describe('ResourceDetail', () => {
  it('renders resource detail page', async () => {
    render(
      <MemoryRouter initialEntries={['/3']}>
        <ResourceDetail
          match={{
            params: {
              id: '3',
            },
          }}
        />
      </MemoryRouter>
    );

    const buttons = await screen.findAllByRole('button', { timeout: 4000 });
    expect(buttons).toMatchSnapshot();

  });
});
