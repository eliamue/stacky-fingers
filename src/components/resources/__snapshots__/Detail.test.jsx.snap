// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceDetail renders resource detail page 1`] = `
Array [
  <button
    name="delete-btn"
    role="button"
  >
    Delete Resource
  </button>,
  <button>
    Edit Resource
  </button>,
]
`;
